### PR TITLE
Fix building on MinGW

### PR DESCRIPTION
**Pull request by MehdiChinoune**

### CODE REVIEW

**Comment by MehdiChinoune** [comment]
> Can you please describe this change in detail?

- `_wopen` expects `wchar_t*` (`wstring`) arguments but `toNative` returns `char*` (`string`), so use `_open` instead on MinGW
- `std::filesystem::path` couldn't be converted to `string`, so use `path::string`
- the arguments of `wmain` are `wchar_t` which couldn't be passed to `formNative` on MinGW.

**Comment by hobu** [comment]
This code does not compile cleanly on all platforms we test with continuous integration. Please update it to ✅ on all three platforms before it can be considered for inclusion.

**Comment by MehdiChinoune** [comment]
@hobu I have pushed new changes, consider approving CI run.

**Comment by MehdiChinoune** [comment]
pushed new changes.

File: untwine/Common.cpp [hunk 1]
```diff
@@ -129,7 +133,11 @@ std::vector<std::string> directoryList(const std::string& dir)
         fs::directory_iterator end;
         while (it != end)
         {
+#ifndef __MINGW32__
```
Comment (abellgithub) [inline untwine/Common.cpp:136]: I think that `string()` is totally sufficient. The macro tests can go away, along with the utwine::fromNative() call. But this needs to be testsed.

File: untwine/Common.cpp [hunk 2]
```diff
@@ -55,7 +55,11 @@ MapContext mapFile(const std::string& filename, bool readOnly, size_t pos, size_
 #ifndef _WIN32
     ctx.m_fd = ::open(filename.data(), readOnly ? O_RDONLY : O_RDWR);
 #else
+#ifdef _MSC_VER
```
Comment (abellgithub) [inline untwine/Common.cpp:58]: I'd appreciate some comments saying what cases are which. If you do MinGW all day, you may know what's what, but for everyone else, these cases are unclear.

File: untwine/Common.cpp (hunk 1 above)
Comment (MehdiChinoune) [reply untwine/Common.cpp:136]: string() doesn't work with MSVC compiler (It asks for wstring())

Comment (abellgithub) [reply untwine/Common.cpp:136]: I will check.

**Comment by MehdiChinoune** [comment]
ping

**Pull request merged**